cdap-ui/app/cdap/components/AbstractWidget/GetSchemaWidget/index.tsx

```diff
@@ -15,27 +15,63 @@
  */
 
 import * as React from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
 import Button from '@material-ui/core/Button';
 import { IWidgetProps } from 'components/AbstractWidget';
 import { objectQuery } from 'services/helpers';
+import IconSVG from 'components/IconSVG';
 
-interface IGetSchemaProps extends IWidgetProps<null> {}
+const styles = (): StyleRules => {
+  return {
+    button: {
+      width: '105px',
+      height: '30px',
+    },
+    spinner: {
+      fontSize: '16px',
+    },
+  };
+};
+
+interface IGetSchemaProps extends IWidgetProps<null>, WithStyles<typeof styles> {}
 
-const GetSchemaWidget: React.FC<IGetSchemaProps> = ({ extraConfig }) => {
+const GetSchemaWidgetView: React.FC<IGetSchemaProps> = ({ extraConfig, classes }) => {
   const validateProperties = objectQuery(extraConfig, 'validateProperties');
+  const [loading, setLoading] = React.useState<boolean>(false);
+
+  function onClickHander() {
+    if (loading) {
+      return;
+    }
+
+    if (validateProperties && typeof validateProperties === 'function') {
+      setLoading(true);
+      validateProperties(() => {
+        setLoading(false);
+      });
+    }
+  }
+
+  const loadingIcon = (
+    <span className={`fa fa-spin ${classes.spinner}`}>
```

Comment:
use classnames?

Reply:
classnames is only useful if there are conditional classes being applied